src/ucp/am/ucp_am.inl

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (C) 2022, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCP_AM_INL_
+#define UCP_AM_INL_
+
+#include <ucp/core/ucp_am.h>
+#include <ucp/core/ucp_request.h>
+
+
+static UCS_F_ALWAYS_INLINE void
+ucp_am_fill_header(ucp_am_hdr_t *hdr, ucp_request_t *req)
+{
+    hdr->am_id         = req->send.msg_proto.am.am_id;
+    hdr->flags         = req->send.msg_proto.am.flags;
+    hdr->header_length = req->send.msg_proto.am.header_length;
+}
+
+static UCS_F_ALWAYS_INLINE void
+ucp_am_pack_user_header(void *buffer, ucp_request_t *req)
+{
+    ucp_dt_state_t hdr_state;
+
+    hdr_state.offset = 0ul;
+
+    ucp_dt_pack(req->send.ep->worker, ucp_dt_make_contig(1),
+                UCS_MEMORY_TYPE_HOST, buffer, req->send.msg_proto.am.header,
+                &hdr_state, req->send.msg_proto.am.header_length);
+}
+
+static UCS_F_ALWAYS_INLINE int
+ucp_am_check_op_id(const ucp_proto_init_params_t *init_params,
+                   ucp_operation_id_t op_id)
+{
+    return (init_params->select_param->op_id == op_id);
+}
+
+static UCS_F_ALWAYS_INLINE int
+ucp_am_check_op_flags(const ucp_proto_init_params_t *init_params,
+                      uint16_t op_flags)
+{
+    return (init_params->select_param->op_flags & op_flags);
+}
```

Comment:
IMO these one-liners do not deserve a special function; in any case, they should be in proto.h or proto_common.h and can use them for all protocols - since they are not really am-specufic